tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/SymbStateDecoder.scala

```diff
@@ -45,7 +45,8 @@ class SymbStateDecoder(solverContext: SolverContext, rewriter: SymbStateRewriter
       } else {
         // if not in the cache, it might be the case that another cell, which has asserted equivalence
         // with the original cell can be found
-        findCellInSet(rewriter.modelValueCache.values().toSeq, cell.toNameEx) match {
+        val values = rewriter.modelValueCache.values().filter(_.cellType == ConstT(uninterpretedType)).toSeq
+        findCellInSet(values, cell.toNameEx) match {
```

Comment:
Why didn't we filter for `cellType` here before? I'm lacking context to understand this change.

Reply:
Looks like that was simply wrong. It was not updated after introducing model values.

Reply:
SG